dev/Repeater/UniformGridLayoutState.cpp

```diff
@@ -119,10 +119,13 @@ void UniformGridLayoutState::SetSize(
     }
 }
 
-void UniformGridLayoutState::EnsureFirstElementOwnership()
+void UniformGridLayoutState::EnsureFirstElementOwnership(winrt::VirtualizingLayoutContext const& context)
 {
-    if (m_flowAlgorithm.GetElementIfRealized(0))
+    if (m_cachedFirstElement != nullptr && m_flowAlgorithm.GetElementIfRealized(0))
     {
+        // We created the element, but then flowlayout algorithm took ownership, so we can clear it and
+        // let flowlayout algorithm do its thing.
+        context.RecycleElement(m_cachedFirstElement);
```

Comment:
This doesn't handle the case where we created the element, but then flowlayout _didn't_ take ownership, in which case the element still needs to be recycled IIUC?

Reply:
We want to keep it cached (and not recycle) in that case to use for the next pass.